doc/dev/release.md

```diff
@@ -327,9 +350,7 @@ The release process for the samples repo is simple:
     $ git push --tags
     ```
 
-### (Post-release) Updating the release notes
-
-Add the following line to the top of the GitHub release notes for `v1.3.0`:
+Then, add the following line to the top of the GitHub release notes for `v1.3.0`:
 
 ```md
 **NOTE:** ensure the `v1.3.0` tag is referenced when referring to sample code in the [SDK Operator samples repo](https://github.com/operator-framework/operator-sdk-samples/tree/v1.3.0) for this release. Links in SDK documentation are currently set to the samples repo `master` branch.
```

Comment:
Add another line that either:
1. Has a link to the section of the version upgrade guide that corresponds to this release.
2. Says that the release is fully backwards compatible and should require no migration steps.

Reply:
This part is speaking about the Release Notes of the samples repo and not over SDK. So, do you think that is really required we add the link for the migration doc on it as well? Should not make sense have it just in the SDK repo? 

Regards: `Says that the release is fully backwards compatible and should require no migration steps.`
A step always will be required which is pinned the SDK new version in the go.mod so IHMO it is not required.

Reply:
This step refers to the SDK's release notes, since we don't have release notes for the samples repo. This should be its own post-release step. The title should be `Updating the Operator SDK release notes` to be explicit about which release notes need updating.

Reply:
Done ;-)